pil-analyzer/src/evaluator.rs

```diff
@@ -821,7 +827,42 @@ impl<'a, 'b, T: FieldElement, S: SymbolLookup<'a, T>> Evaluator<'a, 'b, T, S> {
                         }
                         ta
                     });
-                    self.symbols.lookup(&poly.name, &type_args)?
+
+                    let impl_pos = type_args.as_ref().and_then(|type_args| {
+                        let key = type_args.iter().format(",").to_string();
```

Comment:
Formatting types to string is very expensive. Can you use `Vec<Type>` as a key instead (might have to use BTreeMap, which is fine)?